### PR TITLE
Deny bare trait objects in src/bootstrap

### PR DESCRIPTION
**Pull request by ljedrz**
Enforce `#![deny(bare_trait_objects)]` in `src/bootstrap`.

### CODE REVIEW

**Comment by petrochenkov** [comment]
cc https://github.com/rust-lang/cargo/issues/5034

File: src/bootstrap/lib.rs [hunk 1]
```diff
@@ -113,7 +113,7 @@
 //! More documentation can be found in each respective module below, and you can
 //! also check out the `src/bootstrap/README.md` file for more information.
 
-#![deny(warnings)]
```
Comment (Eijebong) [inline src/bootstrap/lib.rs]: Shouldn't this be kept ?

Comment (ljedrz) [reply src/bootstrap/lib.rs]: I wasn't getting any warnings when I ran my test build without it, so I removed it; was I too quick to do so? I can revert this bit it if I have missed some unpleasant consequence.

Comment (ljedrz) [reply src/bootstrap/lib.rs]: Ah, you're right - I actually misread it as `#![allow(warnings)]` 🙈. I'll revert this bit.

**Pull request merged**